pkg/operator/ceph/object/user/controller.go

```diff
@@ -286,8 +287,35 @@ func (r *ReconcileObjectStoreUser) createorUpdateCephUser(u *cephv1.CephObjectSt
 	// Set access and secret key
 	r.userConfig.Keys[0].AccessKey = user.Keys[0].AccessKey
 	r.userConfig.Keys[0].SecretKey = user.Keys[0].SecretKey
-
 	logger.Info(logCreateOrUpdate)
+
+	if u.Spec.MaxObjects != 0 || u.Spec.MaxSize != "" {
+		quotaEnable := true
+		var maxSize int64 = -1
+		var maxObjects int64 = -1
+		if u.Spec.MaxObjects >= 0 {
+			maxObjects = int64(u.Spec.MaxObjects)
+		}
+		if u.Spec.MaxSize != "" {
+			maxSizeInt, err := resource.ParseQuantity(u.Spec.MaxSize)
```

Comment:
The validations for the webhook should also validate that the quantity parses.

It would also be good to add a helper to the API so you can just cal `u.Spec.GetMaxSize()` and all the error logic will be handled (for both this code and the webhook)